doc/coq.rst

```diff
@@ -142,19 +142,10 @@ The semantics of the fields are:
   You may still use installed libraries in your Coq project, but there is
   currently no way for Dune to know about it.
 
-- You can enable the production of Coq's native compiler object files by setting
-  ``<coq_native_mode>`` to ``native``. This passes ``-native-compiler on`` to
-  Coq and install the corresponding object files under ``.coq-native``, when in
-  the ``release`` profile. The regular ``dev`` profile skips native compilation
-  to make the build faster. This has been available since :ref:`Coq lang
-  0.3<coq-lang>`.
-
-  Please note: support for ``native_compute`` is **experimental** and requires a
-  version of Coq later than 8.12.1. Furthermore, dependent theories *must* be
-  built with the ``(mode native)`` enabled. In addition to that, Coq must be
-  configured to support native compilation. Dune explicitly disables the
-  generation of native compilation objects when ``(mode vo)`` is enabled,
-  irrespective of the configuration of Coq. This will be improved in the future.
+- From version :ref:`Coq lang 0.7<coq-lang>` onwards, the native mode can be
+  automatically detected by Dune by querying ``coqc -config``. You may override
+  this by specifying ``<coq_native_mode>``. Before :ref:`Coq lang
```

Comment:
Maybe
```suggestion
- From version :ref:`Coq lang 0.7<coq-lang>` onwards, the native mode is
  automatically detected by Dune by querying ``coqc -config``. You may override
  this by specifying ``<coq_native_mode>``. Before :ref:`Coq lang
```
or "is, by default,"

Maybe also add a sentence like "You may override this by specifying ``` ``<coq_native_mode>`` ``` with ``` ``(mode native)`` ``` or ``` ``(mode vo)`` ```. The latter can be useful in specific cases of code known not to compile with `` `coqnative <https://coq.inria.fr/refman/practical-tools/utilities.html#split-compilation-of-native-computation-files>`_ ``. Otherwise it is advised to not override the default value."